### PR TITLE
Adding create Namespace support for Metrics-Server addon

### PR DESCRIPTION
**Pull request by tsahiduek**
*Issue #, if available:*
#772 
*Description of changes:*
Add `createNamespace` support for Metrics Server addon

Tests run successfully 
```
Test Suites: 13 passed, 13 total
Tests:       59 passed, 59 total
Snapshots:   0 total
Time:        22.749 s
Ran all test suites.
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/metrics-server/index.ts [hunk 1]
```diff
@@ -1,33 +1,60 @@
...
+        if (this.options.createNamespace == true) {
+            // Let CDK Create the Namespace
+            const namespace = createNamespace(this.options.namespace!, cluster);
+            const chart = this.addHelmChart(clusterInfo, values);
+            chart.node.addDependency(namespace);
+            return Promise.resolve(chart);
+        } else {
+            //Namespace is already created
+            const chart = this.addHelmChart(clusterInfo, values);
+            return Promise.resolve(chart);
```
Comment (elamaran11) [inline lib/addons/metrics-server/index.ts:57]: This statement is duplicated, Please have this outside of the if

File: lib/addons/metrics-server/index.ts [hunk 2]
```diff
@@ -1,33 +1,60 @@
...
 
     constructor(props?: MetricsServerAddOnProps) {
         super({ ...defaultProps, ...props });
-        this.options = this.props;
+        this.options = this.props as MetricsServerAddOnProps;
     }
 
-    deploy(clusterInfo: ClusterInfo): void {
-        this.addHelmChart(clusterInfo, this.options.values);
+    // deploy(clusterInfo: ClusterInfo): void {
```
Comment (elamaran11) [inline lib/addons/metrics-server/index.ts:39]: Remove these commented lines.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@tsahiduek Nice PR. Please make these changes also doc updates are missing.

File: lib/addons/metrics-server/index.ts (hunk 1 above)
Comment (tsahiduek) [reply lib/addons/metrics-server/index.ts:57]: Correct. Did the same for `kube-state-metrics` which I've used as a source

**Comment by tsahiduek** [comment]
> @tsahiduek Nice PR. Please make these changes also doc updates are missing.

Done

File: lib/addons/metrics-server/index.ts [hunk 3]
```diff
@@ -1,33 +1,60 @@
...
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        let values: Values = this.options ?? {};
+        values = merge(values, this.props.values ?? {});
+
+        if (this.options.createNamespace == true) {
+            // Let CDK Create the Namespace
+            const namespace = createNamespace(this.options.namespace!, cluster);
+            const chart = this.addHelmChart(clusterInfo, values);
+            chart.node.addDependency(namespace);
```
[the remark this answers is not in the document]
Comment (tsahiduek) [reply lib/addons/metrics-server/index.ts:52]: yes. It was in `kube-state-metrics`. 
modified it as well

**Review by shapirov103 — APPROVED** [review]
LGTM

**Review by elamaran11 — APPROVED** [review]
LGTM too!

**Pull request merged**